WebPortal/Pages/PowerBI/PowerBIEmbeddedPage.razor

```diff
@@ -31,24 +70,66 @@
     [SupplyParameterFromQuery]
     public bool ReadOnly { get; set; }
 
+    private Guid ReportIdGuid;
+
     private ElementReference PowerBIElement;
 
     private Report report;
     private string reportToken;
 
+    private bool ErrorLoadingReport = false;
+
+    private bool ErrorNotificationSent = false;
+    private bool SendingErrorNotification = false;
+    private SuccessIndicator NotificationSuccessInd;
+
+    private PowerBi_Report dbReport = null;
+
+    private PowerBi_Workspace dbWorkspace => dbReport?.Workspace;
+    private Datahub_Project project => dbWorkspace?.Project;
+    private string projectAcronym => project?.Project_Acronym_CD;
+    private string reportTitle => dbReport?.Report_Name;
+
     protected override async Task OnInitializedAsync()
     {
```

Comment:
I found doing similar pages its safer to load the report OnParametersSet(), instead of OnInitializedAsync, it will ensure all the elements are loaded, specifically the powerbielement.

Reply:
IMO, I prefer the OnInitialized callback, but the OnParameterSet is the correct place to reset state if the component is reusable, which I discourage unless it is a very simple component.

Reply:
In this case I don't think it would make a difference, as the parameters aren't going to change during the page's life cycle, and the PowerBiElement only gets populated after everything is loaded anyway